### PR TITLE
Add stack upgrade docs

### PR DESCRIPTION
**Pull request by anyasabo**
Fix https://github.com/elastic/cloud-on-k8s/issues/1814

This is a first attempt at documenting our ability to upgrade the stack. It's a big feature for us and seemed worthwhile to have a more notable mention. Also should help some related issues such as https://github.com/elastic/cloud-on-k8s/issues/2247 by linking to the upstream docs.

### CODE REVIEW

File: docs/upgrading-stack.asciidoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-upgrading-stack.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-upgrading-stack"]
+=== Upgrading the stack
```
Comment (charith-elastic) [inline docs/upgrading-stack.asciidoc]:
```suggestion
=== Upgrading to newer versions of the Elastic stack
```

File: docs/upgrading-stack.asciidoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+The operator can safely perform upgrades to newer versions of the various Elastic Stack resources.
+
+Note that you will still want to follow the process defined in the link:https://www.elastic.co/guide/en/elastic-stack/current/upgrading-elastic-stack.html#[Elasticsearch documentation]. That is to say, check to ensure that your cluster is compatible with the target version, take a snapshot, and upgrade different resources in the order described (along with any other steps mentioned in the referenced article). Provided you are ready to proceed with the upgrade, all you need to do is modify the `version` field of the resource to the version you would like, and the operator will upgrade it for you according to best practices and minimizing disruption.
```
Comment (charith-elastic) [inline docs/upgrading-stack.asciidoc]:
```suggestion
Note that you should still follow the instructions in the link:https://www.elastic.co/guide/en/elastic-stack/current/upgrading-elastic-stack.html[Elasticsearch documentation] and ensure that your cluster is compatible with the target version, take backups, and follow the specific upgrade instructions for each resource type (especially the order in which the upgrade should be carried out). When you are ready to proceed, simply modify the `version` field in the resource spec to the desired stack version and the operator will start the upgrade process automatically.
```

File: docs/upgrading-stack.asciidoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+The <<{p}-orchestration>> docs include more detail on how the operator performs upgrades and how to tune its behavior.
```
Comment (charith-elastic) [inline docs/upgrading-stack.asciidoc]:
```suggestion
See <<{p}-orchestration>> for more information on how the operator performs upgrades and how to tune its behavior.
```

**Comment by anyasabo** [comment]
Thanks @charith-elastic, those were all very helpful. I updated it if you can take a look again.

**Pull request merged**